Revise libraries for kicad 5

stevefalco opened the pull request:
Please pull to update to version 5 symbols, footprints, and 3d models.

stevefalco committed:
Revise libraries for kicad 5

nickoe commented on kicad.spec.template:
```diff
@@ -28,8 +28,11 @@ URL:            http://www.kicad-pcb.org
 #Source: https://github.com/KiCad/kicad-source-mirror/archive/%{commit}/kicad-source-mirror-%{commit}.tar.gz
 Source1:        %{name}-i18n-%{version}.tar.gz
-Source2:        %{name}-libraries-%{version}.tar.gz
+#Source2:        %{name}-libraries-%{version}.tar.gz
```
Shouldn't this just be removed?

nickoe commented on kicad.spec.template:
```diff
@@ -275,6 +293,9 @@ update-mime-database %{?fedora:-n} %{_datadir}/mime &> /dev/null || :
 
 
 %changelog
```
I reality I think we should just remove this crud, in the modern days we have git.

stevefalco replied:
Done.  I removed the Source2 line, and all of the changelog contents.

stevefalco committed:
Clean up spec file

nickoe commented:
I see you used tabs for indentation. I personally prefer to have expandtab enabled with ts=4. But I will merge anyways, because the formatting is a mess anyways and would require some cleanup separately.

stevefalco commented:
Understood.  I have been working with Aimylios on a more radical change.  He created a spec file that produces separate packages for the libraries, main program, documentation, etc.  That makes it more convenient during development testing, because only the main program is typically updated, while the (huge) libraries can be left in place.

In combination with his spec change, I rewrote the scripts from the ground up to make them more readable and maintainable.  I also added the capability to do a local mock build in addition to the copr build.

I don't know if this gets too far away from what you want for the nightlies, but if you'd like to consider it, I'd be happy to work with you on putting it in place.

You can preview the code on my "release" branch here: https://github.com/stevefalco/fedora-packaging/tree/release

I don't use the URL method of submitting to copr; instead I just upload the SRPM directly.  I don't know if this would have issues for your jenkins builder, so if we were to pursue this, I'd want an account on jenkins, so I could test my changes before attempting to "go live".

Please let me know if this is of any interest.

aimylios commented:
@stevefalco ~~Please remove my email address from your post.~~ (Thanks @nickoe!) I don't want it to be publicly visible. You can reference me by just writing @aimylios.

By the way, you should never use your master branch for PRs. This will cause issues in case your PR is not accepted or needs to be modified. Always create a separate branch per PR and keep your master clean and in sync with the upstream master.

I plan to submit a pull request with some SPEC file cleanup later today, split up in smaller chunks (similar to what I did in my [develop](https://github.com/aimylios/fedora-kicad-packaging/tree/develop) branch). You based your work off my [release](https://github.com/aimylios/fedora-kicad-packaging/tree/release) branch, which is not a suitable base for further pull requests to this repository (at least not regarding the SPEC file).

I suggest that we open an issue here in this repo to discuss the pros and cons of putting the libraries into subpackages. I would prefer to keep shipping a single nightly RPM until all other issues have been resolved.

stevefalco commented:
Good advice about the master branch.  I've used git before, but never on such a collaborative model, so I need to come up to speed on using branches more.  Thanks for the tip.

nickoe commented:
I am fine with whatever change makes this easier to maintain. Also change that make the resulting package more close to what could be upstream to the normal fedora packages are welcomed. I don't use fedora myself for the time being, so I have a harder time testing. I rely on reading the code, checking logs and listening to direct user feedback.

stevefalco commented:
I'll try to reach out to the Fedora maintainer and see what he is thinking.

Do you know how I would go about getting access to jenkins, so I can test my scripts in the real "nightlies" environment?  I don't want to break the nightlies, or cause you extra work.

nickoe commented:
@stevefalco I don't think you need access on jenkins. All the job does is something like this:
```
git fetch
git checkout remotes/origin/master
# clean old src.rpm and tar files to save disk space
rm -rf *.src.rpm
rm -rf *.tar.xz
# go nightlies go!
./nightlies.sh $JOB_URL/ws/
```
If you can build it locally, I guess you can also inspect or install the packages on that machine. Then you don't need to send anything to copr.
If you just submit small and atomic commits in the pull request, they are easy to review and I will probably merge them quickly.

stevefalco commented:
Ok - I'll do more testing, then ask you to pull.

My main concern is with the URL "$JOB_URL/ws/" in the existing jenkins script.  It shouldn't be needed any more, because copr can upload the srpm directly from disk - there is no need to provide a web-based download.  This would be the one piece I would test if I had jenkins access.  I expect it to work, but I tend to be cautious. :-)

We will want to remove the clean-up of the old src.rpm and tar files from the jenkins job, because those files will now be created in build/rpmbuild rather than in the base directory. My scripts automatically delete the old contents of build/rpmbuild prior to making the new build, so there won't be any wasted disk space.  Therefore the jenkins script should be simplified to this:

git fetch
git checkout remotes/origin/master
# go nightlies go!
./nightlies.sh

stevefalco commented:
Argggg.  I guess I got a bit burned by the markup.  Make that:

```
git fetch
git checkout remotes/origin/master
# go nightlies go!
./nightlies.sh
```

nickoe commented:
Yes, that I understand, but it works now, so I has low priority to modify that part for now. The real packaging task is in the actual spec file template. Including if we shall provide libs as separate packages as is done for debian and the PPA. I guess that you can have a multi-package in one spec file.

(You can edit your posts on github)
